packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -257,16 +257,45 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
     }
   }
 
-  /// Whether this route can perform a transition to the given route.
+  /// Returns true if this route supports a transition animation that runs
+  /// when [nextRoute] is pushed on top of it or when [nextRoute] is popped
+  /// off of it.
   ///
   /// Subclasses can override this method to restrict the set of routes they
   /// need to coordinate transitions with.
+  ///
+  /// If true, and `nextRoute.canTransitionFrom()` is true, then the
+  /// [buildTransitions] `secondaryAnimation` will run from 0.0 - 1.0
+  /// when [nextRoute] is pushed on top of this one.  Similarly, if
+  /// the [nextRoute] is popped off of this route, the
+  /// `secondaryAnimation` will run from 1.0 - 0.0.
+  ///
```

Comment:
While you're here, can you document the consequences of it being false?

Reply:
Good point, done.